### PR TITLE
Objects to push to excel.

### LINKED ISSUE #91: Object for excel
#### Description:

As only a single type of objects can be pushed to excel via the excel adapter, there needs to be an object that holds the required data in the SAP toolkit.

### PR DESCRIPTION
**Pull request by EKAdebo**
Closes #91 

This PR should add in a method and an object, to format the relevant data, with the purpose of pushing it to excel. 

### Additional comments

These may need to be updated in the future so that more results can be included.

### CODE REVIEW

File: SAP_Engine/Create/ExcelResults.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+namespace BH.Engine.Environment.SAP
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description(".")]
```
Comment (FraserGreenroyd) [inline SAP_Engine/Create/ExcelResults.cs]: Can we raise an issue to do a proper description

File: SAP_Engine/Create/ExcelResults.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+                TER = energyUseResults.TER,
+
+                DPER = energyUseResults.DPER,
+
+                TPER = energyUseResults.TPER,
+
+                DFEE = energyUseResults.DFEE,
+
+                TFEE = energyUseResults.TFEE
```
Comment (FraserGreenroyd) [inline SAP_Engine/Create/ExcelResults.cs]:
```suggestion
                Iteration = filename,
                WallArea = report.WallArea(),
                WindowArea = report.WindowArea(),
                TFA = report.TotalFloorArea(),
                DER = energyUseResults.DER,
                TER = energyUseResults.TER,
                DPER = energyUseResults.DPER,
                TPER = energyUseResults.TPER,
                DFEE = energyUseResults.DFEE,
                TFEE = energyUseResults.TFEE
```

File: SAP_Engine/Modify/OpeningType.cs [hunk 3]
```diff
@@ -109,6 +109,7 @@ public static SAPReport ModifyOpeningTypes(this SAPReport sapObj, string type, L
                     if (!item.IsNullOrEmpty())
                     {
                         openingObj = openingObj.ModifyOpeningType(item["NewName"]);
+
```
Comment (FraserGreenroyd) [inline SAP_Engine/Modify/OpeningType.cs]:
```suggestion
```

File: SAP_oM/SAPExcelResults.cs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+using System.Linq;
+using System.Xml.Serialization;
+using BH.oM.Base;
+
+namespace BH.oM.Environment.SAP
+{
+    [Description("Details of a shower outlet.")]
+    public class SAPExcelResults : IObject
+    {
+        [Description(".")]
```
Comment (FraserGreenroyd) [inline SAP_oM/SAPExcelResults.cs]: Can we raise an issue to resolve these?

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by EKAdebo** [comment]
@BHoMBot check compliance

**Comment by EKAdebo** [comment]
@BHoMBot check compliance

**Review by FraserGreenroyd — APPROVED** [review]
Changes look good, good work @EKAdebo

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core
@BHoMBot check serialisation

**Pull request merged**